warehouse/query-core/src/test/java/datawave/query/jexl/visitors/JexlFormattedStringBuildingVisitorTest.java

```diff
@@ -0,0 +1,349 @@
+		String query = "((_Value_ = true) && (BAR == 'foo'))";
+		String expected = "((_Value_ = true) && (BAR == 'foo'))\n";
+		JexlNode node = JexlASTHelper.parseJexlQuery(query);
+        String builtQuery = JexlFormattedStringBuildingVisitor.buildQuery(node);
+        
+        Assert.assertEquals(expected, builtQuery);
+	}
+	
+	@Test
+	public void testMarkerNodesSingleTerm2() throws ParseException {
+		// Nested marker node anded with a single term, should be printed on same line
+		String query = "(((_Value_ = true) && (BAR == 'foo')))";
+		String expected = "(((_Value_ = true) && (BAR == 'foo')))\n";
+		JexlNode node = JexlASTHelper.parseJexlQuery(query);
+        String builtQuery = JexlFormattedStringBuildingVisitor.buildQuery(node);
+        
+        Assert.assertEquals(expected, builtQuery);
+	}
+	
+	
+	@Test
+	public void testMarkerNodeAndedMultipleTerms() throws ParseException {
+		// Marker node anded with subtree (not single term or range), should be printed on separate lines
+		String query = "((_Eval_ = true) && (FOO == 'bar' && (BAR == 'foo' || BAR == 'blah')))";
+		String expected = "(\n" + 
+				"	(_Eval_ = true) && \n" + 
+				"	(\n" + 
+				"		FOO == 'bar' && \n" + 
+				"		(\n" + 
+				"			BAR == 'foo' || \n" + 
+				"			BAR == 'blah'\n" + 
+				"		)\n" + 
+				"	)\n" + 
+				")\n";
+		/*
+		 * (
+				(_Eval_ = true) && 
+				(
+					FOO == 'bar' && 
+					(
+						BAR == 'foo' || 
+						BAR == 'blah'
+					)
+				)
+			)
+		 */
+		JexlNode node = JexlASTHelper.parseJexlQuery(query);
+        String builtQuery = JexlFormattedStringBuildingVisitor.buildQuery(node);
+        
+        Assert.assertEquals(expected, builtQuery);
+	}
+	
+	@Test
+	public void testMarkerNodeAndedMultipleTerms1() throws ParseException {
+		// Marker node anded with subtree (not single term or range), should be printed on separate lines
+		String query = "((_Eval_ = true) && (BAR == 'foo' || BAR == 'blah'))";
+		String expected = "(\n" + 
+				"	(_Eval_ = true) && \n" + 
+				"	(\n" + 
+				"		BAR == 'foo' || \n" + 
+				"		BAR == 'blah'\n" + 
+				"	)\n" + 
+				")\n";
+		/*
+		 * (
+				(_Eval_ = true) && 
+				(
+					BAR == 'foo' || 
+					BAR == 'blah'
+				)
+			)
+		 */
+		JexlNode node = JexlASTHelper.parseJexlQuery(query);
+        String builtQuery = JexlFormattedStringBuildingVisitor.buildQuery(node);
+        
+        Assert.assertEquals(expected, builtQuery);
+	}
+	
+	@Test
+	public void testNestedMarkerNodes() throws ParseException {
+		// Nested marker nodes anded with a single term or range, should be printed as a single line
+		String query = "((_Value_ = true) && ((_Bounded_ = true) && (NUM >= 0 && NUM <= 10)))";
+		String expected = "((_Value_ = true) && ((_Bounded_ = true) && (NUM >= 0 && NUM <= 10)))\n";
+		JexlNode node = JexlASTHelper.parseJexlQuery(query);
+        String builtQuery = JexlFormattedStringBuildingVisitor.buildQuery(node);
+        
+        Assert.assertEquals(expected, builtQuery);
+	}
+}
```

Comment:
I haven't looked at the implementation yet, but I ran a few test queries against this.  For the most part, it works really well, but can you add support for this test case as well?  

```suggestion
    @Test
    public void testMultiParensIndentation() throws ParseException {
        String query = "((((FOO == 'BAR' && ((_Eval_ = true) && (THIS == 'THAT'))))))";
        String expected = "((((\n\tFOO == 'BAR' && \n\t((_Eval_ = true) && (THIS == 'THAT'))\n))))\n";
        JexlNode node = JexlASTHelper.parseJexlQuery(query);
        String builtQuery = JexlFormattedStringBuildingVisitor.buildQuery(node);

        Assert.assertEquals(expected, builtQuery);
    }
}
```

Reply:
Thank you for your comments and extra test case! Currently working on fixing the check-code-formatting test failure. I will work to fix my code to use 4 spaces instead of tabs. I will also fix it so there is no new line at the end of the query, and will add support for your given test case.